politeiad/backendv2/tstorebe/plugins/ticketvote/cmds.go

```diff
@@ -1266,8 +1185,66 @@ func (p *ticketVotePlugin) voteColliderSave(token []byte, vc voteCollider) error
 	return p.tstore.BlobSave(token, *be)
 }
 
-// castVoteSave saves a CastVoteDetails to the backend.
-func (p *ticketVotePlugin) castVoteSave(token []byte, cv ticketvote.CastVoteDetails) error {
+// ballotResults is used to aggregate data for votes that are cast
+// concurrently.
+type ballotResults struct {
+	sync.RWMutex
+	addrs   map[string]string                   // [ticket]commitmentAddr
+	replies map[string]ticketvote.CastVoteReply // [ticket]CastVoteReply
+}
+
+// newBallotResults returns a new ballotResults context.
+func newBallotResults() ballotResults {
+	return ballotResults{
+		addrs:   make(map[string]string, 40960),
+		replies: make(map[string]ticketvote.CastVoteReply, 40960),
+	}
+}
+
+// addrSet sets the largest commitment addresss for a ticket.
+func (r *ballotResults) addrSet(ticket, commitmentAddr string) {
+	r.Lock()
+	defer r.Unlock()
+
+	r.addrs[ticket] = commitmentAddr
+}
+
+// addrGet returns the largest commitment address for a ticket.
+func (r *ballotResults) addrGet(ticket string) (string, bool) {
+	r.RLock()
+	defer r.RUnlock()
+
+	a, ok := r.addrs[ticket]
+	return a, ok
+}
+
+// replySet sets the CastVoteReply for a ticket.
+func (r *ballotResults) replySet(ticket string, cvr ticketvote.CastVoteReply) {
+	r.Lock()
+	defer r.Unlock()
+
+	r.replies[ticket] = cvr
+}
+
+// replyGet returns the CastVoteReply for a ticket.
+func (r *ballotResults) replyGet(ticket string) (ticketvote.CastVoteReply, bool) {
+	r.RLock()
+	defer r.RUnlock()
+
+	cvr, ok := r.replies[ticket]
```

Comment:
return

Reply:
`return r.replies[ticket]` only returns a `CastVoteReply`.  I want to return `(CastVoteReply, bool)` so that I can explicity check if the lookup failed.  To do that you need to to do:

```
	cvr, ok := r.replies[ticket]
	return cvr, ok
```